### PR TITLE
refactor checkpointing: remove date and run id

### PR DESCRIPTION
**Pull request by samsja**
this pr makes ckpt easier.

It just put everything in the ckpt folder and the diloco case creates one folder for each diloco worker.

Screenshot from 2024-08-06 12-42-40

tested on a 60m model. The resume done at 100 steps and both colors cannot be distinguished

### CODE REVIEW

File: open_diloco/train_fsdp.py [hunk 1]
```diff
@@ -69,26 +70,18 @@ def log(message):
...
+def check_checkpoint_path_access(checkpoint_path: str, rank: int):
+    dummy_file_path = os.path.join(checkpoint_path, f"dummy_file_{rank}.txt")
     with fsspec.open(dummy_file_path, "w") as f:
         f.write("This is a dummy file for testing access.")
     gfs = GenericFileSystem()
     gfs.rm(dummy_file_path)
 
 
+def get_diloco_rank_path(world_rank_diloco: int):
+    return os.path.join(f"diloco_rank_{world_rank_diloco}")
```
Comment (Jackmin801) [inline open_diloco/train_fsdp.py:82]: Seems like the os.path.join here is unnecessary?

Comment (Jackmin801) [reply open_diloco/train_fsdp.py:82]: its more a folder / directory name than a path. maybe the function name could be different to reflect that.

**Review by Jackmin801 — CHANGES_REQUESTED** [review]
nice! will make resuming a lot cleaner and convenient. added some comments about function naming but otherwise looks great :)

File: open_diloco/train_fsdp.py (hunk 1 above)
Comment (samsja) [reply open_diloco/train_fsdp.py:82]: yeah good point, i was doing an actual join before but I removed it. Will update

**Review by Jackmin801 — APPROVED** [review]
nice! lgtm

**Pull request merged**